Source/WebCore/platform/graphics/cocoa/IOSurface.mm

```diff
@@ -213,6 +216,8 @@
     : m_colorSpace(colorSpace)
     , m_surface(surface)
 {
+    ASSERT(surface);
```

Comment:
Is it possible for this assertion to be hit? Isn't surface null checked before it can reach this code? If so, an assertion might make more sense in the `::createFromSurface` method so it has the chance to be hit while in a debug build.

Reply:
I was going to say "defense mechanism hey!" but this is a private constructor already. I can remove this then. About asserting vs. returning `nullptr`, I like the latter more because `createFromImage` below does that.

Reply:
Yup, it looks like `DisplayBufferDisplayDelegate` expects `createFromSurface` to potentially return a `nullptr` so it makes sense.

Reply:
It might make sense to assert at the end of `DisplayBufferDisplayDelegate::setDisplayBuffer` in order to catch some simple errors, but I am not familiar with this code enough.